nservicebus/upgrades/index.md

```diff
@@ -51,3 +51,10 @@ See also:
  * [NuGet Package Manager Dialog - Updating a Package](https://docs.microsoft.com/en-us/nuget/tools/package-manager-ui#updating-a-package)
  * [NuGet Package Manager Console - Updating a Package](https://docs.microsoft.com/en-us/nuget/tools/package-manager-console#updating-a-package)
 
+## Public API Deprecation
+
+When preparing for a major NServiceBus upgrade, it is crucial to proactively address any potential breaking changes by thoroughly checking your system for warnings and errors at compile time.
+
+As part of the [release policy](/nservicebus/upgrades/release-policy.md#deprecation), APIs are marked as obsolete in a major version before removing them in the next. These obsolete APIs will generate build warnings, providing you with a clear indication of the code that needs to be updated. Treating these warnings as errors and resolving them ensures that your system will not fail at runtime after the upgrade.
```

Comment:
```suggestion
As part of the [release policy](/nservicebus/upgrades/release-policy.md#deprecation), APIs are marked as obsolete, with a warning in a minor release and an error in a major release, before removing them in the next major release. These obsolete APIs will generate build warnings or errors, clearly indicating the code that needs to be updated. Treating warnings as errors and resolving them ensures your system will not fail at runtime after the upgrade.
```

Reply:
FYI we are trying to follow this practice with the minor but this practice is not always possible or feasible to follow. I think it would be better to word this as best effort.